addon/components/polaris-resource-list/bulk-actions.js

```diff
@@ -0,0 +1,535 @@
+      ) {
+        return 'Actions';
+      }
+
+      return 'More actions';
+    }
+  ),
+
+  shouldRenderActionsPopover: computed(
+    'actionSections',
+    'rolledInPromotedActions.length',
+    'measuring',
+    function() {
+      let {
+        actionSections,
+        rolledInPromotedActions,
+        measuring,
+      } = this.getProperties(
+        'actionSections',
+        'rolledInPromotedActions',
+        'measuring'
+      );
+
+      return Boolean(
+        actionSections || rolledInPromotedActions.length > 0 || measuring
+      );
+    }
+  ),
+
+  actionSections: computed('actionsCollection.[]', function() {
+    let actionsCollection = this.get('actionsCollection');
+
+    if (!actionsCollection || actionsCollection.length === 0) {
+      return null;
+    }
+
+    if (this.instanceOfBulkActionListSectionArray(actionsCollection)) {
+      return actionsCollection;
+    }
+
+    if (this.instanceOfBulkActionArray(actionsCollection)) {
+      return [{ items: actionsCollection }];
+    }
+  }),
+
+  combinedActions: computed(
+    'actionSections.[]',
+    'rolledInPromotedActions',
+    function() {
+      let combinedActions = [];
+      let { actionSections, rolledInPromotedActions } = this.getProperties(
+        'actionSections',
+        'rolledInPromotedActions'
+      );
+
+      let rolledInPromotedActionsHasLength = rolledInPromotedActions.length > 0;
+
+      if (actionSections && rolledInPromotedActionsHasLength) {
+        combinedActions = [
+          { items: rolledInPromotedActions },
+          ...actionSections,
+        ];
+      } else if (actionSections) {
+        combinedActions = actionSections;
+      } else if (rolledInPromotedActionsHasLength) {
+        combinedActions = [{ items: rolledInPromotedActions }];
+      }
+
+      return combinedActions;
+    }
+  ),
+
+  /**
+   * Callback when selectable state of list is changed
+   *
+   * @type {Function}
+   * @default noop
+   * @property onSelectModeToggle
+   */
+  onSelectModeToggle(/* selectMode **/) {},
+
+  /**
+   * Callback when the select all checkbox is clicked
+   *
+   * @type {Function}
+   * @default noop
+   * @property onToggleAll
+   */
+  onToggleAll() {},
+
+  instanceOfBulkActionListSectionArray(actionsCollection) {
+    let validList = actionsCollection.filter((action) => {
+      return action.items;
+    });
+
+    return actionsCollection.length === validList.length;
+  },
+
+  instanceOfBulkActionArray(actionsCollection) {
+    let validList = actionsCollection.filter((action) => {
+      return !action.items;
+    });
+
+    return actionsCollection.length === validList.length;
+  },
+
+  setContainerNode() {
+    this.set('containerNode', this.element);
+  },
+
+  setMoreActionsNode(node) {
+    this.set('moreActionsNode', node);
+  },
+
+  setLargeScreenButtonsNode(node) {
+    this.set('largeScreenButtonsNode', node);
+  },
+
+  handleResize() {
+    throttle(
+      this,
+      function() {
```

Comment:
Just want to check that this throttling is working... I know that things like `scheduleOnce` can lose their "once" behaviour if you pass in an anonymous function rather than a method on the object, for example. In this case, each time `handleResize` gets called, `throttle` will be called with a _new_ anonymous function passed in as the callback each time; if `throttle` is using the function pointer to determine if it should run the callback yet, then the throttling behaviour will be lost. I mean, if the throttle implementation is something like this:

```js
throttle(target, callback, ...) {
  if (target === lastTarget && callback === lastCallback) {
    // do the throttling
  } else {
    // start throttle queue for callback
  }
}
```

then you have to pass the _same_ function instance each time you call it for throttling to work. Hope that makes sense 😅